etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/CopyOrcOrderProviderToObrOrderProviderTest.groovy

```diff
@@ -0,0 +1,195 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.custom
+
+import gov.hhs.cdc.trustedintermediary.ExamplesHelper
+import gov.hhs.cdc.trustedintermediary.context.TestApplicationContext
+import gov.hhs.cdc.trustedintermediary.external.hapi.HapiFhirResource
+import gov.hhs.cdc.trustedintermediary.external.hapi.HapiHelper
+import gov.hhs.cdc.trustedintermediary.wrappers.MetricMetadata
+import org.hl7.fhir.r4.model.Bundle
+import org.hl7.fhir.r4.model.Practitioner
+import org.hl7.fhir.r4.model.ServiceRequest
+import spock.lang.Specification
+
+
+class CopyOrcOrderProviderToObrOrderProviderTest extends Specification{
+
+    final String PRACTITIONER_EXTENSION_URL = "https://reportstream.cdc.gov/fhir/StructureDefinition/xcn-practitioner"
+
+    def transformClass
+
+    def setup() {
+        TestApplicationContext.reset()
+        TestApplicationContext.init()
+        TestApplicationContext.register(MetricMetadata, Mock(MetricMetadata))
+        TestApplicationContext.injectRegisteredImplementations()
+
+        transformClass = new CopyOrcOrderProviderToObrOrderProvider()
+    }
+
+    def "when both practitioner resources populated"() {
+        given:
+        final String EXPECTED_NPI = "1790743185"
+        final String EXPECTED_FIRST_NAME = "EUSTRATIA"
+        final String EXPECTED_LAST_NAME = "HUBBARD"
+        final String EXPECTED_NAME_TYPE_CODE = "NPI"
+        final String EXPECTED_IDENTIFIER_TYPE_CODE = null
+        final String FHIR_ORU_PATH = "../CA/007_CA_ORU_R01_CDPH_produced_UCSD2024-07-11-16-02-17-749_1_hl7_translation.fhir"
+
+        def bundle = createBundle(FHIR_ORU_PATH)
+        def serviceRequest = createServiceRequest(bundle)
+
+        expect:
+        evaluateOrc12Values(serviceRequest, EXPECTED_NPI, EXPECTED_FIRST_NAME, EXPECTED_LAST_NAME, EXPECTED_NAME_TYPE_CODE, EXPECTED_IDENTIFIER_TYPE_CODE)
+        evaluateObr16Values(serviceRequest, null, EXPECTED_FIRST_NAME, EXPECTED_LAST_NAME, null, null)
+
+        when:
+        transformClass.transform(new HapiFhirResource(bundle), null)
+
+        then:
+        evaluateOrc12Values(serviceRequest, EXPECTED_NPI, EXPECTED_FIRST_NAME, EXPECTED_LAST_NAME, EXPECTED_NAME_TYPE_CODE, EXPECTED_IDENTIFIER_TYPE_CODE)
+        evaluateObr16Values(serviceRequest, EXPECTED_NPI, EXPECTED_FIRST_NAME, EXPECTED_LAST_NAME, EXPECTED_NAME_TYPE_CODE, EXPECTED_IDENTIFIER_TYPE_CODE)
+    }
+
+    def "when ORC-12 extension populated and OBR-16 extension not populated"() {
```

Comment:
Test name suggestion:
`def "when ORC.12 extension is populated and OBR-16 extension not populated, ORC.12 id copied over"`